src/extension.ts

```diff
@@ -1,98 +1,113 @@
+import { TelemetryEventQueue } from './utils/telemetryEventQueue';
+import { TelemetryService } from './services/telemetryService';
 
-import { TelemetryEvent } from './TelemetryEvent';
-import { TelemetryEventQueue } from './TelemetryEventQueue';
-import { Reporter } from './Reporter';
-import { Logger } from './Logger';
-
-const OPTIN_REQUESTED_KEY = 'redhat.telemetry.optInRequested';
-
+// this method is called when your extension is activated
 export function activate(context: vscode.ExtensionContext) {
-  Reporter.initialize(context);
+  Logger.log('"vscode-commons" is now active!');
 
-  if (!getTelemetryEnabledConfig()) {
+  /* 
+    check for current status of telemetry true | false
+    initialize events queue to preserve events 
+  */
+  if (!TelemetryService.getTelemetryEnabledConfig()) {
     Logger.log('redhat.telemetry.enabled is false');
     TelemetryEventQueue.initialize();
   } else {
     Logger.log('redhat.telemetry.enabled is true');
   }
 
-  openTelemetryOptInDialogIfNeeded(context);
+  TelemetryService.openTelemetryOptInDialogIfNeeded(context);
 
+  context.subscriptions.push(
+    vscode.commands.registerCommand(
+      'vscodeCommons.openWebPage',
+      TelemetryService.openWebPage
+    )
+  );
+
+  /* 
+    listener for configuration change
+  */
   context.subscriptions.push(onDidChangeTelemetryEnabled());
 
+  /* 
+    test command to activate and  view telemetry status of 
+    extension through command pallet can be removed later 
+  */
+  context.subscriptions.push(
+    vscode.commands.registerCommand('vscodeCommons.showTelemetryStatus', () => {
+      vscode.window.showInformationMessage(
+        `Red Hat Telemetry Enabled: ${TelemetryService.getTelemetryEnabledConfig()}`
+      );
+    })
+  );
+
   // This command exists only for testing purposes. Should delete later.
-  context.subscriptions.push(vscode.commands.registerCommand('vscodeCommons.clearStateAndSettings', () => {
-    context.globalState.update('optInRequested', undefined);
-    vscode.workspace.getConfiguration('redhat.telemetry').update('enabled', undefined, true);
-  }));
-
-  context.subscriptions.push(vscode.commands.registerCommand('vscodeCommons.openWebPage', openWebPage));
-
-  if (Reporter.isConnected()) {
-    // export to other extensions
-    return Promise.resolve({
-      reportIfOptIn
-    });
-  }
+  context.subscriptions.push(
+    vscode.commands.registerCommand(
+      'vscodeCommons.clearStateAndSettings',
+      () => {
+        context.globalState.update('OPT_IN_STATUS', undefined);
+        vscode.workspace
+          .getConfiguration('redhat.telemetry')
+          .update('enabled', undefined, true);
+      }
+    )
+  );
+
+  /* 
+    These are the APIs which  are exposed by this extension
+    These APIs can be used in by other extensions as exports
+    Please view INTEGRATION.md for more details
+  */
+  // export to other extensions
+  return Promise.resolve({
+    getTelemetryService,
+    viewMessage,
```

Comment:
getRedHatUUID too

Reply:
@fbricon made changes to support this
PTAL: https://github.com/redhat-developer/vscode-commons/pull/3/commits/25c192608ca6d22ad3239d6e6dd40003ec7d0c9d